src/math.jl

```diff
@@ -97,6 +100,92 @@ Base.sqrt(q::UnionAbstractQuantity) = new_quantity(typeof(q), sqrt(ustrip(q)), s
 Base.cbrt(d::AbstractDimensions{R}) where {R} = d^inv(convert(R, 3))
 Base.cbrt(q::UnionAbstractQuantity) = new_quantity(typeof(q), cbrt(ustrip(q)), cbrt(dimension(q)))
 
-Base.abs(q::UnionAbstractQuantity) = new_quantity(typeof(q), abs(ustrip(q)), dimension(q))
 Base.abs2(q::UnionAbstractQuantity) = new_quantity(typeof(q), abs2(ustrip(q)), dimension(q)^2)
 Base.angle(q::UnionAbstractQuantity{T}) where {T<:Complex} = angle(ustrip(q))
+
+############################## Require dimensionless input ##############################
+# Note that :clamp, :cmp, :sign already work
+# We skip :rad2deg, :deg2rad in case the user defines a rad or deg unit
+for f in (
+    :sin, :cos, :tan, :sinh, :cosh, :tanh, :asin, :acos,
+    :asinh, :acosh, :atanh, :sec, :csc, :cot, :asec, :acsc, :acot, :sech, :csch,
+    :coth, :asech, :acsch, :acoth, :sinc, :cosc, :cosd, :cotd, :cscd, :secd,
+    :sinpi, :cospi, :sind, :tand, :acosd, :acotd, :acscd, :asecd, :asind,
+    :log, :log2, :log10, :log1p, :exp, :exp2, :exp10, :expm1, :frexp, :exponent,
+)
+    @eval function Base.$f(q::UnionAbstractQuantity)
+        iszero(dimension(q)) || throw(DimensionError(q))
+        return $f(ustrip(q))
+    end
+end
+for (type, base_type, _) in ABSTRACT_QUANTITY_TYPES, f in (:atan, :atand)
+    @eval begin
+        function Base.$f(x::$type)
+            iszero(dimension(x)) || throw(DimensionError(x))
+            return $f(ustrip(x))
+        end
+        function Base.$f(y::$type, x::$type)
+            dimension(y) == dimension(x) || throw(DimensionError(y, x))
+            y, x = promote(y, x)
```

Comment:
Probably just need to add `:atan` and `:atand` to the loop below `# Combining different abstract types`